packet.go

```diff
@@ -1020,3 +1044,64 @@ func (p sshFxpExtendedPacketHardlink) respond(s *Server) responsePacket {
 	err := os.Link(p.Oldpath, p.Newpath)
 	return statusFromError(p, err)
 }
+
+type sshFxpExtendedPacketCheckFile struct {
+	ID              uint32
+	ExtendedRequest string
+	HashAlgoList    string
+	StartOffset     uint64
+	Length          uint64
+	BlockSize       uint32
+}
+
+type sshFxpExtendedPacketCheckFileHandle struct {
+	sshFxpExtendedPacketCheckFile
+	Handle string
+}
+
+type sshFxpExtendedPacketCheckFileName struct {
+	sshFxpExtendedPacketCheckFile
+	Path string
+}
+
+func (p sshFxpExtendedPacketCheckFile) id() uint32     { return p.ID }
+func (p sshFxpExtendedPacketCheckFile) readonly() bool { return true }
+func (p *sshFxpExtendedPacketCheckFile) unmarshalBinary(b []byte) (string, error) {
+	var err error
+	var pathOrHandle string
+	if p.ID, b, err = unmarshalUint32Safe(b); err != nil {
+		return pathOrHandle, err
+	} else if p.ExtendedRequest, b, err = unmarshalStringSafe(b); err != nil {
```

Comment:
None of these `else`s are necessary. The other path will always terminate, so nothing is gained by putting every following statement in an `else` clause.